### DISCUSSION TITLE
p-table not displayed in page

### DISCUSSION THREAD

**Discussion by mithilwane** [primefaces/community#3259] [PrimeNG]
Hi,

I have copied the code exactly from primeng ptable examples. But the table refuses to get displayed in the page.. Please help
html
`<div class="row">
        <div class="card-inner row-border">
            <h4 class="fw-bold top-margin">Observations</h4>
            <div>
            <p-table [value]="observationsData" [tableStyle]="{'min-width': '40rem', 'min-height': '10rem'}">
                <ng-template #header>
                    <tr>
                        <th>Sr No</th>
                        <th>Observations</th>
                        <th>Remarks</th>
                    </tr>                    
                </ng-template>
                <ng-template  #body let-observationsData>
                    <tr>
                        <td>{{observationsData.srno}}</td>
                        <td>{{observationsData.observations}}</td>
                        <td>{{observationsData.remarks}}</td>
                    </tr>
                </ng-template>
            </p-table>
        </div>
        </div>        
    </div>   `
ts code `import { Component, Input, OnInit, ViewChild } from '@angular/core';
import { FormBuilder, FormGroup } from '@angular/forms';
import { NgbActiveModal, NgbTypeahead } from '@ng-bootstrap/ng-bootstrap';
import { debounceTime, distinctUntilChanged, filter, map, merge, Observable, OperatorFunction, Subject } from 'rxjs';
import { ObservationsService } from '../../shared/services/observations.service';
import { Observations } from '../../domain/observations';

@Component({
  selector: 'app-repairable-details',
  templateUrl: './repairable-details.component.html',
  styleUrl: './repairable-details.component.scss',
})

export class RepairableDetailsComponent implements OnInit{

  observationsData!: Observations[];

  constructor(private observationService: ObservationsService) { }

  ngOnInit(): void {
    this.observationService.getObservations().then((data) => {
      this.observationsData = data;
    });

  }  
}`

![image](https://github.com/user-attachments/assets/d84d1de6-2baf-4450-adb5-0deee534a32d)

**Comment by mithilwane** [comment]
Got the Issue and Fixed it

**Reply by valentin-baron** [reply]
which was...?

**Reply by Jester0027** [reply]
> which was...?

I had the same issue, I was on angular v18 with primeng v17.
I just upgraded primeng to v18 and it was fixed.

**Reply by Lewis21** [reply]
I had the same issue, To solve with primeng version 17 you have to replace #header with pTemplate=“header” and #body with pTemplate="body” into ng-template
